### PR TITLE
[API Discssion] Add an example API call from the Warnings plugin

### PR DESCRIPTION
**Pull request by uhafner**
Seems to be easier for me too discuss this topic using some real code... 

Not intended to be merged, just a kind of scratch book.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/WarningsChecks.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                .setDescription("Where is this description shown? \n"
+                        + "This string supports [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)");
+
+        // Iterate over all issues in the warnings plugin
+        for (int i = 0; i < 20; i++) {
+            checks.addAnnotation(annotations.setMessage("Message").setLineStart(20).setLevel(WARNING).build());
+        }
+
+        // TODO: how to handle images?
+    }
```
Comment (XiongKezhi) [inline src/main/java/io/jenkins/plugins/WarningsChecks.java:30]: In the end of the method, does it still has to invoke `build` method and return result or delegate these works to the checks plugin?

Comment (XiongKezhi) [reply src/main/java/io/jenkins/plugins/WarningsChecks.java:30]: I think about it a little more and I think I may misunderstand it before.

So now, about this API design, what we need to do is to simply store the values set by setters and then, use these stored values to send requests to GitHub, right?

File: src/main/java/io/jenkins/plugins/WarningsChecks.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package io.jenkins.plugins;
+
+import java.util.logging.Level;
+
+import hudson.model.Run;
+
+import static java.util.logging.Level.*;
+
+// TODO: Do we need an option to report progress during a step (like using the TaskListener interface)?
```
[the remark this answers is not in the document]
Comment (uhafner) [reply src/main/java/io/jenkins/plugins/WarningsChecks.java:9]: Sorry, seems that GitHub emails from this repository are somehow located in a folder where I do not look into on a daily basis... 

Yes, the `TaskListener` is used to log something only.

File: src/main/java/io/jenkins/plugins/WarningsChecks.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+// TODO: Currently 50 annotations are supported per request. Should be also provide a customization?
+public class WarningsChecks {
```
[the remark this answers is not in the document]
Comment (uhafner) [reply src/main/java/io/jenkins/plugins/WarningsChecks.java:11]: Yes, I omitted that for now since your current API does not match.

**Comment by uhafner** [comment]
I think we do not need this PR anymore.

**Pull request closed without merging**